tests/DCPS/DpShutdown/dpshutdown.cpp

```diff
@@ -1,322 +1,395 @@
+    } else {
+      ACE_DEBUG((LM_DEBUG,
+        "Created participant 2 with instance handle %d\n",
+        participant2->get_instance_handle()));
+    }
 
-      if (config.is_nil())
-        {
-          config =
-            OpenDDS::DCPS::TransportRegistry::instance()->create_config(config_1.c_str());
-        }
+    try {
+      OpenDDS::DCPS::TransportRegistry::instance()->bind_config(config2, participant2.in());
+    } catch (const OpenDDS::DCPS::Transport::MiscProblem&) {
+      ACE_ERROR_RETURN((LM_ERROR, "%N:%l: main() ERROR: TransportRegistry::bind_config() throws "
+        "Transport::MiscProblem exception\n"), 1);
+    } catch (const OpenDDS::DCPS::Transport::NotFound&) {
+      ACE_ERROR_RETURN((LM_ERROR, "%N:%l: main() ERROR: TransportRegistry::bind_config() throws"
+        " Transport::NotFound exception\n"), 1);
+    }
 
-      OpenDDS::DCPS::TransportInst_rch inst =
-        OpenDDS::DCPS::TransportRegistry::instance()->get_inst(instance_1.c_str());
+    if (participant1->get_instance_handle() == participant2->get_instance_handle()) {
+      ACE_ERROR_RETURN((LM_ERROR, "%N:%l: main() ERROR: "
+        "participant1 and participant2 do have the same instance handle!\n"), 1);
+    }
 
-      if (inst.is_nil())
-        {
-          inst =
-            OpenDDS::DCPS::TransportRegistry::instance()->create_inst(instance_1.c_str(),
-                                                                      transport.c_str());
+    // Register TypeSupport (Messenger::Message)
+    Messenger::MessageTypeSupport_var mts = new Messenger::MessageTypeSupportImpl();
+    if (mts->register_type(participant1.in(), "") != DDS::RETCODE_OK) {
+      ACE_ERROR_RETURN((LM_ERROR, "%N:%l: main() ERROR: register_type failed!\n"), 1);
+    }
 
-          config->instances_.push_back(inst);
+    // Create Topic
+    CORBA::String_var type_name = mts->get_type_name();
+    DDS::Topic_var topic1 = participant1->create_topic("Movie Discussion List",
+      type_name.in(),
+      TOPIC_QOS_DEFAULT,
+      DDS::TopicListener::_nil(),
+      OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+    if (!topic1) {
+      ACE_ERROR_RETURN((LM_ERROR, "%N:%l: main() ERROR: create_topic topic1 failed!\n"), 1);
+    }
 
-          OpenDDS::DCPS::TransportRegistry::instance()->global_config(config);
-        }
+    // Create Publisher
+    DDS::Publisher_var pub = participant1->create_publisher(
+      PUBLISHER_QOS_DEFAULT, DDS::PublisherListener::_nil(), OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+    if (!pub) {
+      ACE_ERROR_RETURN((LM_ERROR, "%N:%l: main() ERROR: create_publisher failed!\n"), 1);
+    }
 
-      // Create another transport instance for participant2 since RTPS transport instances
-      // cannot be shared by domain participants.
-      OpenDDS::DCPS::TransportConfig_rch config2 =
-        OpenDDS::DCPS::TransportRegistry::instance()->get_config(config_2.c_str());
+    // Create DataWriter
+    DDS::DataWriter_var dw = pub->create_datawriter(topic1.in(),
```

Comment:
```suggestion
    DDS::DataWriter_var dw = pub->create_datawriter(topic1,
```

If we're going to "re-style" I'd suggest taking out `.in()` everywhere it's not needed, making `_nil()` just `0` and not use `DEFAULT_STATUS_MASK` when there's no listener.

Certainly not required though.